src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,80 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int INDEX_OPERATION = 0;
+    private static final int INDEX_DATA = 1;
+    private static final String LIMIT_FIELD = ",";
+    private static final String OPEERATION_ARRIVAL = "supply";
+    private static final String OPERATION_EXPENSE = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String inputInfo;
+        String reportText;
+        inputInfo = readFile(fromFileName);
+        reportText = generateReport(inputInfo);
+        wtiteToFile(reportText, toFileName);
+    }
+
+    private String readFile(String fromFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            int value = reader.read();
+            while ((value != - 1)) {
+                stringBuilder.append((char) value);
+                value = reader.read();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file " + fromFileName, e);
+        }
+        return stringBuilder.toString();
+    }
+
+    private String generateReport(String inputInfo) {
+        if (inputInfo.length() == 0) {
+            return null;
+        }
```

Comment:
returning null silently is a bad practice, better throw an exception if we don't expect this situation. In this task, we don't need to validate input